### PR TITLE
added error responses

### PR DESCRIPTION
**Pull request by Snurre1**
Added some responses to the swagger documentation.
PR to main as users branch is deleted.

### CODE REVIEW

**Review by Melisa-Zorraindo — APPROVED** [review]

File: src/modules/user/userSwagerDocs.js [hunk 1]
```diff
@@ -1,478 +1,499 @@
...
+ *         type: jsonToken
+ *         in: header
+ *         scheme: bearer
+ *     parameters:
+ *       - in: path
+ *         name: id
+ *         required: true
+ *         description: String ID of the user to retrieve.
+ *         schema:
+ *           type: String
```
Comment (Anclagen) [inline src/modules/user/userSwagerDocs.js:395]: We have that the id is required but we need the request body to be defined too for updating a user this was forgotten the first time around.
```js
 *     requestBody:
 *       required: true
 *       content:
 *         application/json:
 *           schema:
 *             type: object
 *             properties:
 *               email:
 *                 type: string (email)
 *                 description: The user's email.
 *                 example: example@example.com
 *               password:
 *                 type: string
 *                 description: The user's desired password.
 *                 example: Password123
 *               confirmpassword:
 *                 type: string
 *                 description: The user's current password.
 *                 example: Password123
 *               etc.......
```

File: src/modules/user/userSwagerDocs.js [hunk 2]
```diff
@@ -1,478 +1,499 @@
...
+ *               properties:
+ *                 data:
+ *                   type: object
+ *                   properties:
+ *                     message:
+ *                       type: string
+ *                       description: Success message.
+ *                       example: Account deleted.
+ *       500:
+ *         description: Internal server error
```
Comment (Anclagen) [inline src/modules/user/userSwagerDocs.js:496]: The updates to delete added some 401 errors
https://github.com/NoroffFEU/agency-api.noroff.dev/blob/main/src/modules/user/controllers/controllerDelete.js

File: src/modules/user/userSwagerDocs.js [hunk 3]
```diff
@@ -1,478 +1,499 @@
...
+ *                       format: date
+ *                       description: Date user joined.
+ *                       example: 2023-01-16T19:55:13.609Z
+ *                     Updated:
+ *                       type: string
+ *                       format: date
+ *                       description: Most recent update.
+ *                       example: 2023-01-16T19:55:13.609Z
+ *       500:
+ *         description: Internal server error
```
Comment (Anclagen) [inline src/modules/user/userSwagerDocs.js:460]: Once these changes get merged we will have more errors on put
https://github.com/NoroffFEU/agency-api.noroff.dev/blob/Users-Testing-Errors/src/modules/user/controllers/controllerUpdate.js

**Review by Anclagen — CHANGES_REQUESTED** [review]
Could you add the request body for PUT and add some of the new error messages that are merged in or going to be merged in.

File: src/modules/user/userSwagerDocs.js (hunk 1 above)
Comment (Snurre1) [reply src/modules/user/userSwagerDocs.js]: done

File: src/modules/user/userSwagerDocs.js (hunk 2 above)
Comment (Snurre1) [reply src/modules/user/userSwagerDocs.js:507]: dont think I added all of the 401`s as the errors did the same, or should I add all of them?

File: src/modules/user/userSwagerDocs.js (hunk 3 above)
Comment (Snurre1) [reply src/modules/user/userSwagerDocs.js:469]: done

**Comment by Fermain** [comment]
Is this ready to merge? Can we resolve conversations that are finished

**Review by Anclagen — APPROVED** [review]

**Pull request merged**